### DISCUSSION TITLE
Soft Delete Middleware and changing return query types

### DISCUSSION THREAD

**Discussion by amcjen** [prisma/prisma#7174] [Q&A]
Hey there-

Following the example [here](https://www.prisma.io/docs/concepts/components/prisma-client/middleware/soft-delete-middleware) on getting soft delete working. So far all works fine, and I'm using the deletedAt timestamp method, not the boolean method.

The problem arises when I attempt to update a record--the middleware triggers successfully, but the return type of the query result model doesn't match up, and I get Prisma type errors.

Upon deeper digging, I'm finding that this soft delete middleware casts the return type to `AffectedRowsOutput`.  There is a note in the soft delete docs that suggests this may be the case, with a con being "Modifying queries in middleware can have some unintended consequences, such as changing query return types".

Question is, how does one preserve the model type for all soft delete middleware calls? Or do I need to use option 1, and implement all soft delete filters in application code, explicitly defining `({ where: { deletedAt: null }})` everywhere I call the queries?

Thank you!

**Comment by janpio** [comment]
cc @mhwelander
